### ISSUE TITLE
Suggestion - Registering a property with the Omni Foundation

### ISSUE THREAD

**Issue by zathras-crypto** [OmniLayer/spec#297]
Hey guys,

I've been reflecting on certain issues and have come up with a suggestion that may serve to address several topics to various degrees in one swoop.

The concept is a simple one.  Registration of a property with the foundation offers zero technical benefits and is not in any way required to use the functionality of the Omni Protocol, however it does grant certain social and marketable benefits via products and services the Omni Foundation sponsors.

The proposed benefits can be aligned with needs we are starting to see entrench themselves in day to day ops.  Such as:
- Addressing tokens as SPT#3126 is rough and quite frankly a horrible UX (send me 50 SPT#565 ugh, try testeco haha send me 50 SPT#275599432 double ugh).  We've looked at multiple options, and I myself have proposed a couple of others (such as floating algo for issuing a SYMBol on property creation) but we haven't really reached a consensus on method.  Registering a property with the foundation would include registration of a SYMBol which would be used in place of the SPT# to display tokens.
- We have no concept of icons for the different tokens in any of the foundation sponsored platforms.  Registration of a property would include submission of an icon/coin image/whatever eg 64x64px for displaying tokens.
- We are limited to 255 characters for property 'detailed information' without sending the client to a web resource (which is not desirable for privacy reasons).  Registration of a property would include (example) 5000 characters additional text and one additional banner image for a property.

Note it's critical to understand that registering a property with the Omni Foundation is not mandatory, and only grants these benefits to _Foundation sponsored_ platforms (so OmniWallet, OmniChest, Omni Core etc) and does not guarantee any 3rd party usage of these resources.

Note, once registered the resources supplied (icons, text, SYMBol etc, if accepted) would be included in the official github repos as part of the source (we would not store them web-side or blockchain-side).

Note, this has no impact at all on the sponsoring a property transaction methodology or validity, it is an entirely centralized concept about the issuer-foundation relationship and the benefits there-in; it bears no technical impact on the technology.  It is simply an ability to allow issuers to register resources (icon/SYMBol etc) for usage in reference software (and _our_ web wallet/block explorer).  I really want to stress that this not about the software of the protocol, it's simply about presentation (and perhaps a little monetization :P)

There are many other similar types of benefits I could harp on about but I think you get the idea.  A potential revenue stream is always helpful and I think it would allow us to really neaten up the UI for properties that want to register (unregistered props would function just fine, just wouldn't be as 'pretty').

Thoughts?

Thanks
Z

**Comment by dacoinminster** [comment]
Interesting idea. So, instead of assigning a human-readable name to every
asset, we'd wait for them to pay us?

**Comment by zathras-crypto** [comment]
> Interesting idea. So, instead of assigning a human-readable name to every asset, we'd wait for them to pay us?

Well, asset creation remains completely unchanged.  For example MaidSafe would still be issued with the human readable name 'MaidSafeCoin (#3)' and function as per current regardless of registration.  There are no protocol changes here at all, only 'display' changes for Omni sponsored systems (Omniwallet, OmniCore, Omnichest etc).

Perhaps I could use 'Chest as an example.  By default, properties will have a generic icon, a generic token **SYMB**ol and no extended property info, which all become available after registration.

**NON-FOUNDATION-REGISTERED PROPERTY**
![image](https://cloud.githubusercontent.com/assets/5633351/5593308/1aeec3d0-9258-11e4-87ae-bb1d07d232db.png)
![image](https://cloud.githubusercontent.com/assets/5633351/5593312/45de75ea-9258-11e4-89e1-c1191cc3111a.png)
![image](https://cloud.githubusercontent.com/assets/5633351/5593328/5982fe30-9259-11e4-8681-aa7e7aedffe2.png)

**FOUNDATION-REGISTERED PROPERTY**
![image](https://cloud.githubusercontent.com/assets/5633351/5593304/e8caf338-9257-11e4-83ff-fe795cff0582.png)
![image](https://cloud.githubusercontent.com/assets/5633351/5593320/a0835574-9258-11e4-87c7-6fc4b3c56468.png)
![image](https://cloud.githubusercontent.com/assets/5633351/5593343/ddb0e6ee-925a-11e4-9f8e-4c783b8d9ed5.png)

TL:DR; registration is a layer above the protocol - nothing is changed on the blockchain for a property registration, we simply add extra data for the property into a new token customization DB which is shipped with each of our sponsored platform.

**Comment by dexX7** [comment]
I think this is great and is sort of related to #250 - moving meta-data off-chain, while this topic is about providing enhanced meta data as a service. This again, is somewhat close to the initial idea of the "promote property" idea, I think.

I have a rather strong opinion on meta-data in general and I'd prefer to get rid of it in general, so a property is only defined and identified by it's id - which, by the way, should not be derivded from previous state (see the problems the P2SH rollout created where property ids where off  by two numbers for old clients). Different topic though, ...

Now I hope there will be the day when Masterchest.info finally becomes *the* explorer for MSC properties again and I think *you* should provide such a service anyway, either with or without consensus or doing a likewise move with other "official products", because it adds value - namely easier property identification and enhanced information.

I have no insight in the whole sponsoring process, but regarding this idea here I have one rather important question:

Who is eligible to register for enhanced property information and such a service?

**Comment by zathras-crypto** [comment]
Thanks for the feedback...

> which, by the way, should not be derivded from previous state (see the problems the P2SH rollout created where property ids where off by two numbers for old clients). Different topic though, ...

May be different topic but I would love to see us revisit this in open discussion with everyone - there are some potential roadblocks we can avoid if we shifted away from sequential property ID's.

> Now I hope there will be the day when Masterchest.info...

FYI masterchest.info is not long for this world - will be taken over by Chest v5 (parses via Core), been spending Xmas trying to get it ready before 0.0.9 tag (STO support etc).  v5 will be launched at http://omnichest.info (very much WIP) to fit in with the rebranding.  Re services from 'Chest, at the moment the foundation sponsors it by paying the hosting fees thus I wouldn't feel right monetizing it personally (though I reserve the right to monetize 'Chest at some point in future if the foundation decides to stop sponsoring for any reason haha!)

> Who is eligible to register for enhanced property information and such a service?

In my view - anyone - you just need to be able to sign a message from the issuing address for the property you want to register to prove ownership, but other than that I see no reason to put in obstacles (though some kind of approval process I guess is needed to avoid accepting logo/text containing something nefarious that'll get us into trouble (eg dodgy pictures of kids or something))

**Comment by dexX7** [comment]
> May be different topic but I would love to see us revisit this in open discussion with everyone

Great to hear, would really like to make progress in this direction. #280 was an attempt in this direction which quickly stalled.

> In my view - anyone

Here is the problem I see: it's for free when using other products (e.g. http://www.blockscan.com/assetInfo/LTBCOIN), but the more relevant issue: "premium" properties imply a certainly higher degree of credibility. If it's free for anyone, there is no exploit vector, because chances are equal and it's "fair", but paid premium properties may open the doors for scam. Let me exaggerate: what if I create a fake MaidSafe token and start to promote it to sell my fake tokens on the DEx? This introduces two consequences:

- It would be a bad outcome, if, for some reason, a scam makes it through the approval process. The process itself is undefined at this point, but I would assume such a process is not trivial and the auditor is probably going to be pitchforked, if something dodgy slips through.

- It may create pressure on legit issuers, because "no premium" properties imply a lesser degree of credibility.

**Comment by zathras-crypto** [comment]
> Great to hear, would really like to make progress...

Made a suggestion in that thread, let me know your thoughts

> Here is the problem I see...

Whilst I don't necessarily disagree with the points raised, I think those are points relevant to the concept of property promotion as a whole rather than the methodology used to provide it.  Whether an issuer pays the foundation to register a property (indirectly providing development funding) or pays the protocol and has the funds burned, the core concept is still the same - that of a token issuer paying to display their property more prominently than the rest.  From the spec:

*Say that having created your "Quantum Miner" smart property (which was assigned property ID #8) you now want it to show up higher in the list of properties. You decide to spend 3 Mastercoins to promote your smart property so that it is displayed higher in the list than all the spam/scam/experimental properties.*

It's essentially the same thing, only difference is where the money goes - my suggestion adds a layer of human approval (whatever form the vetting process takes) which is more than a protocol-only driven approach.  Additionally I think the entire cycle would result in 'real' issuers registering to secure their SYMBol.  With your example of MaidSafeCoin for example, that could only work if MaidSafeCoin hadn't already registered (since you would not be able to re-register the MAID token).

Interesting points though, food for thought :)

**Comment by dexX7** [comment]
> With your example of MaidSafeCoin for example, that could only work if MaidSafeCoin hadn't already registered

It's a non-issue for MaidSafeCoin for several reasons, but might be much more complicated in other cases. 

> I think those are points relevant to the concept of property promotion

I support the idea, and this topic in general, and I agree, my concerns are somewhat related to "property promotion" in general.

**Comment by CraigSellars** [comment]
I am also in favor of this idea.  I like that it would be available in Foundation-supported apps first, but would like to see a JSON format queriable blob that others could pull if they wanted to use that information as well.  Making it generic enough such that other “registrars” would be able to publish info would be good too, but I like the direction this is going in its most simple form.

**Comment by zathras-crypto** [comment]
> would like to see a JSON format queriable blob

Foundation web resources (OmniChest, OmniWallet) can easily provide registered property metadata over an API.

> Making it generic enough such that other “registrars” would be able to publish

Would ship with Core, so anyone using Core can republish this same info :)
